pkg/cmd/create/create_env.go

```diff
@@ -320,7 +320,13 @@ func (o *CreateEnvOptions) RegisterEnvironment(env *v1.Environment, gitProvider
 				return err
 			}
 			sr, err := kube.GetOrCreateSourceRepository(jxClient, devNs, gitInfo.Name, gitInfo.Organisation, gitInfo.HostURLWithoutUser())
-			log.Logger().Debugf("have SourceRepository: %s\n", sr.Name)
+			if err != nil {
+				return errors.Wrap(err, "getting or creating source repository")
+			} else if sr == nil {
```

Comment:
What do you think about using `!=` instead of `==` for all the checks? Keeps the code base consistent. Thoughts? May be also eliminate the need for an `else if` statement?

Reply:
I might be missing something, but I wanted to check 2 things: 1) that there are errors (err != nil) and, if so return the error with a message and 2) if there are no errors but for some reason sr is still nil, log that as a warning, since it seems like an unexpected behavior. Then, the "success" case, if there are no errors and sr has a value, print the source repository name safely.

Reply:
Oh, I think I now understood what you meant. Swap the content in my second and third conditions and replace the else if to be "!="? To me the current approach seems more natural to understand the condition, but I guess it's a matter of preference. My personal vote is to leave it as is in the PR, but if there are strong opinions in favor of your suggestions, I can make the change.

Reply:
1. :+1: 
2. Is there a case where this can be nil (I glanced at the function, dont see where it returns nil ...)? Also, if this is nil, should we not return an error? @abayer @romenrg

Reply:
In this particular case, I don't think we could ever hit a nil without an error, but I'd keep this in to be safe.

Reply:
should we return an error, or just log sr being nil (if it is harmless) - new to this part of the codebase ... ?

Reply:
IMHO the current approach is the safest bet and an improvement over what was already there. In the unlikely event that there is no error but sr is nil, logging the case will allow whoever encounters it to decide if a further improvement must be done to this piece of the codebase or not (depending on if it causes issues later in the code); but by having this log at least he/she will see a message explaining what happened, instead of a nil pointer exception.

Reply:
I am ok with it :+1: